runner/execer/os/os_execer.go

```diff
@@ -89,21 +90,35 @@ func (p *osProcess) monitorMem(memCap execer.Memory, stat stats.StatsReceiver) {
 	reportThresholds := []float64{0, .25, .5, .75, .85, .9, .93, .95, .96, .97, .98, .99, 1}
 	memTicker := time.NewTicker(10 * time.Millisecond)
 	defer memTicker.Stop()
+	// Clean up after ourselves
+	pgid, err := syscall.Getpgid(pid)
```

Comment:
I suspect that this should be done at the top where we do "pid:=". Or we should confirm that if pid exits, all of the children processes that share its pgid have exited as well, but that seems unlikely.